tonic/models/vic/grid_params.py

```diff
@@ -665,60 +881,127 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
 
         out_dicts[name] = out_dict
 
-    if veglib_dict and version == '5.0.dev':
-        # adjust vars for the following conditions
-        # bare soil tile
+    # Merge information from veglib and veg and transfer to veg_dict
+    if veglib_dict and veg_dict:
 
-        # Add bare soil tile
+        # Check for any missing Cv areas that will become bare soil
         var = 'Cv'
         bare = 1 - out_dicts['veg_dict'][var].sum(axis=0)
         bare[bare < 0.0] = 0.0
-        nveg_clases = out_dicts['veg_dict'][var].shape[0] + 1
-        shape = (nveg_clases, ) + out_dicts['veg_dict'][var].shape[1:]
+
+        # Determine the final number of veg classes, accounting for
+        # potential new bare soil class, and determine that class's idx
+        var = 'Cv'
+        if lib_bare_idx >= 0:
```

Comment:
how about `if lib_bare_idx is not None` where the default value (set above) is `None`?